Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs

```diff
@@ -285,12 +285,16 @@ private static void RefreshGeneratedOutput(bool forceGenerateEverything, bool fo
                 MSBuildProjectBuilder.TryBuildAllProjects(MSBuildProjectBuilder.CleanProfileName);
             }
 
-            bool doesTokenFileExist = File.Exists(TokenFilePath);
+            // Get the token file in the Unity Temp directory, if it exists.
+            Version tokenVerison = GetCurrentTokenVersion();
+
+            // We regenerate, if the token file exists, and it's current version.
+            bool doesCurrentVersionTokenFileExist = tokenVerison != null && tokenVerison == MSBuildForUnityVersion;
```

Comment:
Nit: Bring the comment above into the regenerate everything if... comment down below, just so that we can keep the logic + english associated with it close together.

Also I'm having a little trouble parsing the boolean logic -> english - it feels like the comment doesn't match the code i.e.:

We will regenerate everything if !doesCurrentVersionTokenFileExist -> expand out this boolean logic and we regenerate if:

!(tokenVersion != null && tokenVersion == MSB4UVersion) =>
tokenVersion == null || tokenVersion != MSB4UVersion =>
tokenVersion != MSB4UVersion

Reply:
Updated.